src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3171,16 +3171,40 @@ PropertySymbol addEqualityContract()
                                                                      isStatic: false,
                                                                      ImmutableArray<PropertySymbol>.Empty);
 
-                // https://github.com/dotnet/roslyn/issues/44903: Check explicit member has expected signature.
+                PropertySymbol equalityContract;
+
                 if (!memberSignatures.TryGetValue(targetProperty, out Symbol? existingEqualityContractProperty))
```

Comment:
FYI, added a note to test plan to confirm nullability expectations on this signature (if I implement `protected virtual Type? EqualityContract { get; }`, should the compiler give a warning?)